src/vcpkg/cmakevars.cpp

```diff
@@ -137,7 +156,9 @@ endmacro()
 
 function(vcpkg_get_tags PORT FEATURES VCPKG_TRIPLET_ID VCPKG_ABI_SETTINGS_FILE)
     message("d8187afd-ea4a-4fc3-9aa4-a6782e1ed9af")
+    print_variables("15fe3605-2429-46b9-b6b1-c5c008621f79")
     vcpkg_triplet_file(${VCPKG_TRIPLET_ID})
+    print_variables("9dbc61d9-e274-40b4-b38a-4d2d3fe91341")
```

Comment:
I think you also need to probe the environment before and after the triplet.

Reply:
Maybe, but for example you don't want to have the `PATH` in the abi. Another solution would be that you have to set a local variable to the value of the environment variable. 

Do you have an example in which a environment variable should be hashed?

Reply:
You need to include whatever the triplet sets. So the only way to do that is comparing before/after and filtering unrelevant variables out.

Reply:
A problem would be that if a user sets a env variable outside of the triplet and uses this variable inside a port, this would be not caught (But this is probably impossible to detect. Except we include all env vars in the hash, but that would make the binarycache useless). 

> You need to include whatever the triplet sets. So the only way to do that is comparing before/after and filtering unrelevant variables out.

And if someone appends something to the PATH we should only hash the newly added stuff. 

And if I look at 
https://github.com/Neumann-A/my-vcpkg-triplets/blob/99d32a61fe890db22cc668fb4327891aa42f2699/armabi-v7a.cmake#L6
we should remove `${CMAKE_CURRENT_LIST_DIR}` from the variable values before hashing.

Reply:
https://github.com/microsoft/vcpkg/pull/15053/files#diff-e1605373cd9d4822d5c5db0dbeef7008e0657895640b0639f4a3ca5e72d40407 has code to extract changes in the environment from a batch script. You basically need something similar

Reply:
Yeah I can use the code I already have. What I mean is that for example here:
https://github.com/Neumann-A/my-vcpkg-triplets/blob/99d32a61fe890db22cc668fb4327891aa42f2699/x64-windows-llvm.port.cmake#L22
you only want `${POSSIBLE_LLVM_BIN_DIR}` gets hashed and not the whole `ENV{PATH}`

Reply:
And `POSSIBLE_LLVM_BIN_DIR` gets hashed anyways. Hm, do we have an example where it would be really necessary to hash changes to the env?

Reply:
Maybe we need something similar to `VCPKG_ENV_PASSTHROUGH_UNTRACKED`. 

Something like `VCPKG_TRIPLET_VARS_UNTRACKED` and `VCPKG_TRIPLET_ENV_VARS_(UN)TRACKED` to explicitly enable/disable tracking of some (env) vars. Currently vcpkg just tracks everything from the triplet 


> really necessary to hash changes to the env?

Setting any variable which is related to building but not detected by the compiler detection: `AR/ASM/FC`